File: src/main/java/grauly/hudcompass/KeyBindings.java
package grauly.hudcompass;

import grauly.hudcompass.screens.ConfigureWaypointScreen;
import grauly.hudcompass.screens.waypointlist.WaypointListScreen;
import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents;
import net.minecraft.client.Minecraft;
import net.minecraft.client.KeyMapping;
import com.mojang.blaze3d.platform.InputConstants;
import net.minecraft.resources.Identifier;
import org.lwjgl.glfw.GLFW;

public class KeyBindings {
    private static final KeyMapping.Category MAIN = KeyMapping.Category.register(Identifier.fromNamespaceAndPath(HudCompass.MODID, "main"));
    private static final KeyMapping NEW_WAYPOINT = new KeyMapping(
            "key.hudcompass.newpoint",
            InputConstants.Type.KEYSYM,
            GLFW.GLFW_KEY_B,
            MAIN
    );
    private static final KeyMapping WAYPOINT_LIST = new KeyMapping(
            "key.hudcompass.waypointlist",
            InputConstants.Type.KEYSYM,
            GLFW.GLFW_KEY_V,
            MAIN
    );

    private static void tick(Minecraft client) {
        if (NEW_WAYPOINT.consumeClick()) {
            client.gui.setScreen(new ConfigureWaypointScreen(client.gui.screen()));
        }
        if (WAYPOINT_LIST.consumeClick()) {
            client.gui.setScreen(new WaypointListScreen(client.gui.screen()));
        }
    }

    public static void init() {
        ClientTickEvents.END_CLIENT_TICK.register(KeyBindings::tick);
    }
}

File: src/client/kotlin/grauly/hudcompass/KeyBindings.kt
package grauly.hudcompass

import grauly.hudcompass.screens.ConfigureWaypointScreen
import grauly.hudcompass.screens.waypointlist.WaypointListScreen
import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents
import net.minecraft.client.MinecraftClient
import net.minecraft.client.option.KeyBinding
import net.minecraft.client.util.InputUtil
import net.minecraft.util.Identifier
import org.lwjgl.glfw.GLFW

object KeyBindings {
    val MAIN: KeyBinding.Category = KeyBinding.Category.create(Identifier.of(HudCompass.MODID, "main"))
    val NEW_WAYPOINT: KeyBinding = KeyBinding(
        "key.hudcompass.newpoint",
        InputUtil.Type.KEYSYM,
        GLFW.GLFW_KEY_B,
        MAIN
    )
    val WAYPOINT_LIST: KeyBinding = KeyBinding(
        "key.hudcompass.waypointlist",
        InputUtil.Type.KEYSYM,
        GLFW.GLFW_KEY_V,
        MAIN
    )

    private fun tick(client: MinecraftClient) {
/*
        if (NEW_WAYPOINT.wasPressed()) {
            client.setScreen(ConfigureWaypointScreen(client.currentScreen))
        }
        if (WAYPOINT_LIST.wasPressed()) {
            client.setScreen(WaypointListScreen(client.currentScreen))
        }
*/
    }

    fun init() {
        ClientTickEvents.END_CLIENT_TICK.register(ClientTickEvents.EndTick { this::tick })
    }
}